stingray/deadtime/filters.py

```diff
@@ -0,0 +1,237 @@
+import warnings
```

Comment:
I just realized from your comment on the PR that we now have two modules called `filters.py`, one in the top-level `stingray` repo and here. I think this might be confusing? Could we add the code here simply to `events.py`? This would solve the confusion of having two modules of the same name and also the issue of circular imports. :)

Reply:
Ok, I think I've done everything you suggest here, in a slightly different way to avoid circular imports and other complications.